src/ucs/sys/string.h

```diff
@@ -32,7 +32,7 @@ BEGIN_C_DECLS
 
 /* value which specifies "auto" for a numeric variable */
 #define UCS_MEMUNITS_AUTO   ((size_t)-2)
-#define UCS_ULUNITS_AUTO    ((size_t)-2)
+#define UCS_ULUNITS_AUTO    ((unsigned long)-2)
```

Comment:
should we fix `ucs_config_*_ulunits()` functions where we do casting to `size_t*` and `UCS_ULUNITS_INF` macro where we do casting to `size_t`?

Reply:
fied